scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/TestMarkedAsFinished.java

```diff
@@ -0,0 +1,143 @@
+
+import com.jayway.awaitility.Duration;
+import functionaltests.job.error.TestTaskRestartOnNodeFailure;
+import functionaltests.utils.SchedulerFunctionalTestWithCustomConfigAndRestart;
+import functionaltests.utils.SchedulerTHelper;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobStatus;
+
+import org.ow2.proactive.scheduler.common.job.TaskFlowJob;
+import org.ow2.proactive.scheduler.common.task.JavaTask;
+import org.ow2.proactive.scheduler.common.task.OnTaskError;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+import org.ow2.proactive.scheduler.util.FileLock;
+
+
+/**
+ * This test checks that when we mark failed task as finished,
+ * Scheduler does not release node, which is busy with other task.
+ */
+public class TestMarkedAsFinished extends SchedulerFunctionalTestWithCustomConfigAndRestart {
+
+    private static final URL failingJob = TestMarkedAsFinished.class.getResource("/functionaltests/descriptors/Job_failing.xml");
+
+    @BeforeClass
+    public static void startDedicatedScheduler() throws Exception {
+        schedulerHelper = new SchedulerTHelper(false,
+                new File(SchedulerTHelper.class.getResource("/functionaltests/config/scheduler-nonforkedscripttasks.ini")
+                        .toURI()).getAbsolutePath(),
+                null,
+                null);
+        schedulerHelper.createNodeSource("local", 1);
+    }
+
+    @Test
+    public void test() throws Throwable {
+
+        JobId failingJobId = schedulerHelper.submitJob(new File(failingJob.toURI()).getAbsolutePath());
+
+        // task failure will make whole job paused. Because onTaskError="pauseJob"
+        await().timeout(Duration.ONE_MINUTE)
+               .until(silent(() -> schedulerHelper.getSchedulerInterface().getJobState(failingJobId).getStatus()),
+                equalTo(JobStatus.PAUSED));
+
+        final FileLock fileLock = new FileLock();
+        fileLock.lock();
+
+        // submit normal job which should execute for a quite long time
+        JobId normalJobId = schedulerHelper.submitJob(createJob(fileLock.toString()));
+
+        // wait until normal job is running
+        schedulerHelper.waitForEventJobRunning(normalJobId);
+
+        assertEquals(JobStatus.PAUSED, schedulerHelper.getSchedulerInterface().getJobState(failingJobId).getStatus());
+        assertEquals(JobStatus.RUNNING, schedulerHelper.getSchedulerInterface().getJobState(normalJobId).getStatus());
+
+        // after we mark as finished failed task, failing job should become FINISHED
+        // however, even more important, that other job is continue to run
+//        schedulerHelper.getSchedulerInterface().finishInErrorTask(failingJobId.value(), "Error_Task");
+        Thread.sleep(10000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)